File: dotnet/source/amp.bus.rabbit/topology/RouteInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace amp.bus.rabbit.topology
{
    public struct RouteInfo
    {
        public Exchange ProducerExchange { get; private set; }
        public Exchange ConsumerExchange { get; private set; }


        public RouteInfo(Exchange producerExchange, Exchange consumerExchange)
             : this()
        {
            this.ProducerExchange = producerExchange;
            this.ConsumerExchange = consumerExchange;
        }
    }
}

File: java/amp-rabbit/src/main/java/amp/rabbit/topology/RouteInfo.java
package amp.rabbit.topology;


public class RouteInfo {

    protected Exchange consumerExchange;
    protected Exchange producerExchange;


    public RouteInfo() {

    }

    public RouteInfo(Exchange producerExchange, Exchange consumerExchange) {
        this.producerExchange = producerExchange;
        this.consumerExchange = consumerExchange;
    }


    public Exchange getConsumerExchange() {
        return consumerExchange;
    }

    public Exchange getProducerExchange() {
        return producerExchange;
    }

    public void setConsumerExchange(Exchange exchange) {
        consumerExchange = exchange;
    }

    public void setProducerExchange(Exchange exchange) {
        producerExchange = exchange;
    }


    @Override
    public String toString() {

        StringBuilder sb = new StringBuilder("{");

        sb.append(" \"producerExchange\" : ");
        sb.append(producerExchange.toString());
        sb.append(", \"consumerExchange\" : ");
        sb.append(consumerExchange.toString());
        sb.append("}");

        return sb.toString();
    }
}
